sub/sd_ass.c

```diff
@@ -307,8 +307,16 @@ static bool check_packet_seen(struct sd *sd, int64_t pos)
 
 static void decode(struct sd *sd, struct demux_packet *packet)
 {
+    struct mp_subtitle_opts *opts = sd->opts;
     struct sd_ass_priv *ctx = sd->priv;
     ASS_Track *track = ctx->ass_track;
+
+    // Currently no supported text sub formats support a distinction between forced
+    // and unforced lines, so we just assume everything's unforced and discard everything.
+    // If we ever see a format that makes this distinction, we can add support here.
+    if (opts->forced_subs_only_current)
+        return;
```

Comment:
That's questionable.
1. If the option is toggled, the packets are missing
2. Shouldn't this take container flags for forced sub tracks into account?

Reply:
Moved this to `get_bitmaps`, so toggling the option now behaves correctly. Re: 2, this matches the behavior in the bitmap sub case for `sd_lavc.c`, and the track-level flags are handled at a higher level, so I think this is fine?

Reply:
Bitmap subs have a per-event "forced" flag, text subtitles do not. I was thinking that some files have some text subtitle tracks marked as "forced" as a whole, and should be treated as if every event was marked as forced.
Since you're explicitly touching the forced subtitle logic, that seems like something to consider. But if you say this is fine, fine.